src/Model.php

```diff
@@ -1562,7 +1651,7 @@ public function save(array $data = [])
             if ($this->idField && $this->reloadAfterSave) {
                 $d = $dirtyRef;
                 $dirtyRef = [];
-                $this->reload();
+                $this->tryReload();
```

Comment:
no try - we want strict behaviour as much as possible, not to silently ignore possible issues

Reply:
This is not a possible issue but a designed situation - if you do not have any weak condition then we could think about throwing an exception and not null.

Reply:
I do not understand this, please provide a testcase for it.

Reply:
I placed the tryReload for the usecase above (the crud remove unread message from scope on save) - so when a model saves something, the associated view gets a feedback if the entity is still there (in scope, so not null) or if it is gone - so save function only *tries* to reload, and if it is not successful it should return null.

Reply:
I understand, but we cannot remove this check because of some need.

I have no solution is my head, maybe we need to add some support to ui for it.

In atk4/data, we have currently BEFORE_SAVE and AFTER_SAVE hooks, maybe we need some concept of try/finally SAVE hook.

In the past we discussed an immutable models. I am still not fully decided if a condition removal should be supported at all. I have no plan to disallow it in the near future, but to support the conditions removal fully, we need to implement https://github.com/atk4/data/issues/662 first.

If you want to pursue this feature, please start with a real Crud test in atk4/ui. If you need one or two atk4/data classes modified there, you can extend the atk4/data classes in atk4/ui. Please keep in mind, this feature should work also for other ui components like Multiline.

Reply:
Why not rund a proper reload if no weak condition exists - only if a weak condition exists, you only try to reload and return null of it left the scope.

Reply:
Because we need to be able to rely on reload. If the record became out of scope, it can be some error as well, with `tryReload`, we can be left with old data.